mantidimaging/gui/test/gui_system_spectrum_test.py

```diff
@@ -56,3 +57,78 @@ def test_add_roi(self) -> None:
             self.assertEqual(final_roi_count, initial_roi_count + 1)
             self.assertIn(f'roi_{i}', self.spectrum_window.roi_table_model.roi_names())
             self.assertIn(f'roi_{i}', self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_remove_roi(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+        initial_roi_count = self.spectrum_window.roi_table_model.rowCount()
+
+        QTest.mouseClick(self.spectrum_window.removeBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+        final_roi_count = self.spectrum_window.roi_table_model.rowCount()
+
+        self.assertEqual(final_roi_count, initial_roi_count - 1)
+        self.assertNotIn(f'roi_{initial_roi_count - 1}', self.spectrum_window.roi_table_model.roi_names())
+        self.assertNotIn(f'roi_{initial_roi_count - 1}', self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_change_roi_color(self):
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+
+        roi_name = 'roi_1'
+        new_color = (255, 0, 0, 255)
+
+        sensible_roi = SensibleROI(left=0, top=0, right=10, bottom=10)
+        self.spectrum_window.spectrum_widget.add_roi(sensible_roi, roi_name)
+
+        with mock.patch('mantidimaging.gui.windows.spectrum_viewer.spectrum_widget.SpectrumROI.openColorDialog',
+                        return_value=new_color):
+            self.spectrum_window.spectrum_widget.roiColorChangeRequested.emit(roi_name, new_color)
```

Comment:
This can take another step back and call `onChangeColor`. It needed a tweak to the mocking so that the return value was a QColor.
```python
        with mock.patch('mantidimaging.gui.windows.spectrum_viewer.spectrum_widget.SpectrumROI.openColorDialog',
                        return_value=QColor(*new_color)):
            self.spectrum_window.spectrum_widget.roi_dict[roi_name].onChangeColor()
```

You could also use `mock.patch.object` because we have a reference to the thing we are patching.
```python
        spec_roi = self.spectrum_window.spectrum_widget.roi_dict[roi_name]
        with mock.patch.object(spec_roi, 'openColorDialog', return_value=QColor(*new_color)):
            spec_roi.onChangeColor()
```